src/main/java/dev/openfeature/sdk/MutableContext.java

```diff
@@ -92,18 +91,25 @@ public EvaluationContext merge(EvaluationContext overridingContext) {
             return new MutableContext(this.asMap());
         }
 
-        Map<String, Value> merged = new HashMap<String, Value>();
+        Map<String, Value> merged = this.merge(map -> new MutableStructure(map),
+                                               this.asMap(),
+                                               overridingContext.asMap());
 
-        merged.putAll(this.asMap());
-        merged.putAll(overridingContext.asMap());
-        EvaluationContext ec = new MutableContext(merged);
+        String newTargetingKey = "";
 
         if (this.getTargetingKey() != null && !this.getTargetingKey().trim().equals("")) {
-            ec.setTargetingKey(this.getTargetingKey());
+            newTargetingKey = this.getTargetingKey();
         }
 
         if (overridingContext.getTargetingKey() != null && !overridingContext.getTargetingKey().trim().equals("")) {
-            ec.setTargetingKey(overridingContext.getTargetingKey());
+            newTargetingKey = overridingContext.getTargetingKey();
+        }
+        
+        EvaluationContext ec = null;
+        if (newTargetingKey != null && !newTargetingKey.trim().equals("")) {
+            ec = new MutableContext(newTargetingKey, merged);
+        } else {
+            ec = new MutableContext(merged);
```

Comment:
@javicv can you add one more test, similar to the other, which doesn't have a `targetingKey`, to cover this case?

Reply:
@toddbaert I hope we should look into this scenario or will do one more PR? https://github.com/open-feature/java-sdk/issues/270#issuecomment-1421175077

Reply:
> @javicv can you add one more test, similar to the other, which doesn't have a `targetingKey`, to cover this case? I'm OK for merge after this.

New test added

> @toddbaert I hope we should look into this scenario or will do one more PR? [#270 (comment)](https://github.com/open-feature/java-sdk/issues/270#issuecomment-1421175077)

@toddbaert, as I mentioned in [this comment in #270](https://github.com/open-feature/java-sdk/issues/270#issuecomment-1422240459), the current implementation doesn't merge lists, only structures. IMHO, merging lists is a different issue, I can take care of it, but we need to clarify if we want to merge lists or simply, as it's now, override the existing list with the new one.

Reply:
I don't think lists should be merged at all. Merging nested structures is a common use case. I would not expect lists/sets to be merged, but overwritten.